open_spiel/python/mfg/algorithms/distribution.py

```diff
@@ -105,11 +106,21 @@ def evaluate(self):
             new_mfg_state = mfg_state.child(action)
             new_mfg_state_str = new_mfg_state.observation_string(
                 pyspiel.PlayerId.DEFAULT_PLAYER_ID)
-            if new_mfg_state_str not in self.distribution:
+            # As a state can be the child of two different parent states, we do
+            # not add it in the new states if it has already been seen in this
+            # iteration.
+            if new_mfg_state_str not in new_distribution:
               new_listing_states.append(new_mfg_state)
-            self.distribution[new_mfg_state_str] += prob * self.distribution[
-                mfg_state.observation_string(
-                    pyspiel.PlayerId.DEFAULT_PLAYER_ID)]
+            new_distribution[new_mfg_state_str] += prob * self.value(mfg_state)
+
+      assert all(state_str not in self.distribution
+            for state_str in new_distribution), (
+          "Some new states have the same string representation of old states.")
```

Comment:
Can an example be printed, without complicating the code too much?

Reply:
Done.